assets/src/edit-story/components/panels/link.js

```diff
@@ -116,16 +98,19 @@ function LinkPanel({ selectedElements, onSetProperties }) {
   );
 
   useEffect(() => {
-    if (state.link?.url) {
-      populateMetadata(state.link?.url);
+    if (state?.url) {
+      populateMetadata(state?.url);
+    } else if (state.url === '' || state.desc || state.icon) {
+      setState({ url: null, desc: null, icon: null });
+      onSetProperties({ link: null });
     }
-  }, [populateMetadata, state]);
+  }, [onSetProperties, populateMetadata, state]);
```

Comment:
Ok, I'm getting rather stressed about the `useEffect(() => {}, [..., state])` pattern. That's what caused the infinite loop last time, no?